### PR TITLE
feat(paywall): move Paywall init logic to paywall-app

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This PR aims at achieving 2 things: 

- simplify the `@unlock-protocol/paywall` lib by having it directly a `Paywall` object (without taking care of initializing it)
- update the `paywall-app` demo accordingly by initializing the new `Paywall` object correctly

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
If this is too messy maybe we can move oaywall-app to vite completely? As long as we have a way to redirect from the / and host the unlock.min.js file that should be fine?

File: packages/paywall/package.json [hunk 1]
```diff
@@ -29,14 +35,14 @@
...
-    "jest": "27.5.1",
     "jest-fetch-mock": "3.0.3",
     "style-loader": "3.3.1",
     "ts-jest": "27.1.5",
     "ts-loader": "9.4.2",
     "typescript": "4.9.5",
-    "webpack": "5.75.0",
-    "webpack-cli": "5.0.1"
+    "vite": "4.1.4",
+    "vitest": "^0.29.2",
```
Comment (julien51) [inline packages/paywall/package.json:44]: Please pin :)

File: packages/paywall/package.json [hunk 2]
```diff
@@ -29,14 +35,14 @@
...
+    "vitest-fetch-mock": "^0.2.2"
```
Comment (julien51) [inline packages/paywall/package.json:45]: same same

File: packages/paywall/src/init.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { Paywall } from './Paywall'
```
Comment (julien51) [inline packages/paywall/src/init.ts:1]: Would it make sense to move to `paywall-app` rather?

File: paywall-app/src/components/UnlockPaywall.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      }
+
+      window.unlockProtocolConfig = rawConfig
+
+      if (!rawConfig) {
+        console.error(
+          'Missing window.unlockProtocolConfig. See docs on how to configure your locks: https://docs.unlock-protocol.com/'
+        )
+      } else {
+        // set network based on hostname if missing in rawConfig
```
Comment (julien51) [inline paywall-app/src/components/UnlockPaywall.js:27]: Ho I see it is here!

**Review by julien51 — APPROVED** [review]
LGTM

**Comment by clemsos** [comment]
need to rebase when #11257 is merged

**Comment by clemsos** [comment]
still lack 2 things at that point: 

1. backward compat for the default cjs export (we need same behaviour when using directly a `script` tag)
2. some tests failing w vitest in `src/__tests__/utils/optimisticUnlocking.test.js`

**Comment by clemsos** [comment]
> If this is too messy maybe we can move oaywall-app to vite completely? As long as we have a way to redirect from the / and host the unlock.min.js file that should be fine?

we should host `unlock.min.js` in `paywall.unlock-protocol.com/unlock.min.js`, right?

**Comment by searchableguy** [comment]
> backward compat for the default cjs export (we need same behaviour when using directly a script tag)

we shouldn't use init in package.json exports at all. Only for the unlock.min.js cdn file.

File: paywall-app/vite.config.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// vite is only used to compile the `lib/cdn.js` file
+import { resolve } from 'path'
+import { defineConfig } from 'vite'
+
+export default defineConfig({
+  build: {
+    lib: {
+      entry: resolve(__dirname, 'lib/cdn.js'),
+      fileName: 'unlock.latest.min',
+      formats: ['cjs'],
```
Comment (searchableguy) [inline paywall-app/vite.config.js:10]: Should be umd.

I think esm should be the way to go but I suspect that will break

File: paywall-app/src/components/UnlockPaywall.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+          getUserAccountAddress,
+          getState,
+        })
+
+        loadCheckoutModal()
+      }
+    }
+  }, [])
+
+  return <p>Hello world</p>
```
Comment (clemsos) [inline paywall-app/src/components/UnlockPaywall.js:57]:
```suggestion
  return <div />
```

File: paywall-app/src/components/UnlockPaywall.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+export default function UnlockPaywall() {
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const rawConfig = {
+        locks: {
+          '0x0a2173b20e476ce53bcff430f9dbda596d84b9d7': {
+            network: 5,
+          },
+        },
+      }
```
[the remark this answers is not in the document]
Comment (julien51) [reply paywall-app/src/components/UnlockPaywall.js:18]: Yeah I am not sure where this is coming from ... but it should not be here for sure.

**Comment by julien51** [comment]
I will test tghis locally now!

**Comment by julien51** [comment]
```
yarn run dev
warn  - Port 3000 is in use, trying 3001 instead.
ready - started server on 0.0.0.0:3001, url: http://localhost:3001
info  - Defining routes from exportPathMap
error - unable to locate '/Users/julien/repos/unlock/dist/unlock.latest.min.js' glob
wait  - compiling...
error - unable to locate '/Users/julien/repos/unlock/dist/unlock.latest.min.js' glob
wait  - compiling...
error - unable to locate '/Users/julien/repos/unlock/dist/unlock.latest.min.js' glob
wait  - compiling...
error - unable to locate '/Users/julien/repos/unlock/dist/unlock.latest.min.js' glob

```

File: paywall-app/src/components/UnlockPaywall.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (julien51) [reply paywall-app/src/components/UnlockPaywall.js]: Cleaned up!

File: packages/paywall/src/index.ts [hunk 8]
```diff
@@ -1,32 +1,4 @@
...
-  const { getState, getUserAccountAddress, loadCheckoutModal, resetConfig } =
-    paywall
-
-  setupUnlockProtocolVariable({
-    loadCheckoutModal,
-    resetConfig,
-    getUserAccountAddress,
-    getState,
-  })
-}
```
Comment (julien51) [inline packages/paywall/src/index.ts:32]: This part has moved to paywall-app

**Comment by julien51** [comment]
Ok I think it is all good. I will merge once CI has rully run so you all can test this on staging in the next few days!

**Pull request merged**